_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,867 @@
+
+Sign in to GitHub and create a public repository `jhipster-k8s`. Follow the instructions to push the existing `kubernetes` folder from your local using the command line.
+
+**IMPORTANT NOTE**: At this point, you will be pushing plain secrets contained in the application and Kubernetes configuration. To avoid this insecure anti-pattern, you can run the JHipster registry locally to encrypt application configuration, and also set up `kubeseal` for Kubernetes secrets encryption. The process is described in a previous post [Kubernetes to the Cloud with Spring Boot and JHipster](/blog/2021/06/01/kubernetes-spring-boot-jhipster).
+
+Go back to Spinnaker UI, and in the pipeline configuration choose **Add Stage** and set the following values:
+
+- Type: **Deploy(Manifest)**
+- Stage Name: **deploy application-configmap**
+- Account: **jhipster-gke-account**
+- Override Namespace: **yes**
+- Namespace: **demo**
+- Manifest Source: **Artifact**
+- Manifest Artifact: **Define a new artifact**
+
+ In the artifact fields, set the following values:
+
+ - Account: **your-github-username**
+ - Content URL: https://api.github.com/repos/your-github-username/jhipster-k8s/contents/registry-k8s/application-configmap.yml
+ - Commit/Branch: **main**
+
+ Click **Save Changes**.
+ Repeat the process for adding stages to deploy `jhipster-registry.yml`, which must depend on the previous manifest. Also repeat for `gateway-postgresql.yml`, `store-mongodb.yml` manifests, which both should depend on the `jhipster-registry` stage.
+
+Add a stage for the `store-deployment.yml` manifest. For this manifest, you must bind the Docker image from the context. In the deploy manifest configuration, in **Required Artifacts to Bind**, choose **store-image**. The option must be available, as the image was set as an artifact constraint in the trigger configuration, making the artifact available in the pipeline context.
+
+{% img blog/jhipster-ci-cd/sp-bind-artifact.png alt:"Spinnaker deploy manifest configuration binding artifact" width:"700" %}{: .center-image }
+
+Also add a stage for the `gateway-deployment.yml`, binding the `gateway` Docker image. Finally, add stages for `store-service.yml` and `gateway-service.yml`. Notice Docker images have to be bound only for the deployment manifests, where the images are referenced.
+The complete pipeline must look like this:
+
+{% img blog/jhipster-ci-cd/sp-jhipster-pipeline.png alt:"Spinnaker pipeline for jhipster microservices" width:"800" %}{: .center-image }
+
+Test the pipeline manually once. In the pipelines view, click **Start Manual Execution** for the _store-cd_ pipeline. Set the following options:
+
+- Trigger: leave **docker-account: your-dockerhub-username/store**
+- Type: **Tag**
+- Tag: **latest**
+
+Successful execution will show all stages in a green state:
+
+{% img blog/jhipster-ci-cd/sp-pipeline-success.png alt:"Spinnaker pipeline successful execution" width:"800" %}{: .center-image }
```

Comment:
I get an error here about the jhipster-registry not deploying.

<img width="970" alt="Screen Shot 2022-08-09 at 11 59 54" src="https://user-images.githubusercontent.com/17892/183726437-4cbf6cea-1023-4304-bb03-92b76a738c33.png">

I checked the `deploy jhipster-registry` stage and it all looks right. Any ideas?

My YAML file: https://github.com/mraible/jhipster-k8s/blob/main/registry-k8s/jhipster-registry.yml

<img width="1072" alt="Screen Shot 2022-08-09 at 11 59 33" src="https://user-images.githubusercontent.com/17892/183726663-0038c59b-6874-4a04-ac5f-ba56c1b642b9.png">

Reply:
I believe this all failed because I used the wrong URL for the content URL. I'll try changing it to:

```
https://api.github.com/repos/mraible/jhipster-k8s/contents/registry-k8s/jhipster-registry.yml
```

Reply:
I changed all the Content URLs to use `api.github.com`. Now my `deploy-gateway` stage fails after 10 minutes. Do you think it could be a timeout issue?

<img width="823" alt="deploy-gateway" src="https://user-images.githubusercontent.com/17892/183739049-be5434ac-74b6-48f5-bc75-8312336d882d.png">

If I look at `deploy-store`, it has similar warnings but does complete successfully.

<img width="827" alt="deploy-store" src="https://user-images.githubusercontent.com/17892/183739124-81ffc63d-fb5e-4cef-880c-d5d7e66816d6.png">

Reply:
Hi Matt, my gateway also failed when generated with latest JHipster, I left a note in leankit. I was taking too long with the post so I went back to JHIpster 7.8.1 and the gateway starts.

Reply:
If you build the Docker image with the prod profile, it works.